sklearn/tree/_splitter.pyx

```diff
@@ -0,0 +1,1794 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+
+# Authors: Gilles Louppe <g.louppe@gmail.com>
+#          Peter Prettenhofer <peter.prettenhofer@gmail.com>
+#          Brian Holt <bdholt1@gmail.com>
+#          Noel Dawe <noel@dawe.me>
+#          Satrajit Gosh <satrajit.ghosh@gmail.com>
+#          Lars Buitinck <L.J.Buitinck@uva.nl>
+#          Arnaud Joly <arnaud.v.joly@gmail.com>
+#          Joel Nothman <joel.nothman@gmail.com>
+#          Fares Hedayati <fares.hedayati@gmail.com>
+#          Jacob Schreiber <jmschreiber91@gmail.com>
+#
+# Licence: BSD 3 clause
+
+from libc.stdlib cimport free, realloc, qsort
+from libc.string cimport memcpy, memset
+
+import numpy as np
+cimport numpy as np
+np.import_array()
+
+from scipy.sparse import csc_matrix
+
+from ._utils cimport log, safe_realloc, rand_int, rand_uniform
```

Comment:
Normally, we do one import per line. 

>This is better to have one import per line as it's easier to see changes whenever you use a software like git.

It's not consistent everywhere as it's essential.